src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,100 +16,129 @@
+    Long professorRating = Integer.parseInt(request.getParameter("rating-professor"));
     boolean translateToEnglish = Boolean.parseBoolean(request.getParameter("languages"));
 
     if (translateToEnglish) {
-      Translate translateService = TranslateOptions.getDefaultInstance().getService();
-      Translation translationClassFeedback =
-          translateService.translate(classFeedback, Translate.TranslateOption.targetLanguage("en"));
-      Translation translationProfessorFeedback =
-          translateService.translate(
-              professorFeedback, Translate.TranslateOption.targetLanguage("en"));
-      String translatedClassFeedback = translationClassFeedback.getTranslatedText();
-      String translatedProfessorFeedback = translationProfessorFeedback.getTranslatedText();
-      classFeedback = translatedClassFeedback;
-      professorFeedback = translatedProfessorFeedback;
+      termFeedback = translateFeedback(termFeedback);
+      professorFeedback = translateFeedback(professorFeedback);
     }
 
-    Document classFeedbackDoc =
-        Document.newBuilder()
-            .setContent(classFeedbackDoc)
-            .setType(Document.Type.PLAIN_TEXT)
-            .build();
-    Document professorFeedbackDoc =
-        Document.newBuilder()
-            .setContent(professorFeedback)
-            .setType(Document.Type.PLAIN_TEXT)
-            .build();
-
-    LanguageServiceClient classLanguageService = LanguageServiceClient.create();
-    Sentiment classSentiment =
-        classLanguageService.analyzeSentiment(classFeedbackDoc).getDocumentSentiment();
-    float classScore = classSentiment.getScore();
-    classLanguageService.close();
-
-    LanguageServiceClient profLanguageService = LanguageServiceClient.create();
-    Sentiment professorSentiment =
-        profLanguageService.analyzeSentiment(professorFeedbackDoc).getDocumentSentiment();
-    float professorScore = professorSentiment.getScore();
-    profLanguageService.close();
-
-    Entity professorEntity = new Entity("Professor");
-    professorEntity.setProperty("comments-professor", professorFeedback);
-    professorEntity.setProperty("score-professor", professorScore);
-    professorEntity.setProperty("perception", professorRating);
-
-    Entity commentsEntity = new Entity("Comments");
-    commentsEntity.setProperty("comments-class", classFeedback);
-    commentsEntity.setProperty("score-class", classScore);
-    commentsEntity.setProperty("perception", classRating);
+    float termScore = getSentimentScore(termFeedback);
+    float professorScore = getSentimentScore(professorFeedback);
 
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    datastore.put(commentsEntity);
-    datastore.put(professorEntity);
+    // Gets user email.
+    String userId = request.getParameter("ID");
+    // Gets term key from Course object.
+    Key currentTermKey = request.getParameter("Course").term;
```

Comment:
Note: Rework this due to the fact that we're switching to passing each variable (name, prof, term, units) separately in the URL instead of passing the whole Course var